client/src/App.js

```diff
@@ -10,27 +10,87 @@ import NavBar from './components/NavBar/NavBar.js';
 import Fetch from './components/Fetch/Fetch.js';
 
 class App extends Component {
-  render() {
-    return (
-      <div className="App">
-        <nav className="App-navigation">
-          <h1 className="App-title">Trash App</h1>
-          <Link to="/">Welcome</Link>
-          <Link to="/browse/">Browse</Link>
-          <Link to="/recycle/">Recycle</Link>
-        </nav>
-
-        <div className="App-mainContent">
-          <Switch>
-            <Route exact path='/' component={Welcome} />
-            <Route exact path='/browse/' component={Browse} />
-            <Route exact path='/recycle/' component={RecyclePage} />
-          </Switch>
-        </div>
-
-      </div>
-    );
+
+  state = {
+    userLng: -122.269883,
+    userLat: 37.806767,
+    userZoom : 12,
+    geolocation:false,
+    geolocationModal: false
+  }
+
+
+openGeoLocationModal = () => {
+    this.setState({geolocationModal: true})
+}
+
+
+closeGeoLocationModal = () => {
+    this.setState({geolocationModal: false})
+}
+
+
+catchGeoLocationError = (error) => {
+  console.log('error fcn')
+  switch(error.code) {
+    case error.PERMISSION_DENIED:
+      this.openGeoLocationModal()
+      break;
+    case error.POSITION_UNAVAILABLE:
+      this.openGeoLocationModal()
+      break;
+    case error.TIMEOUT:
+      console.log("Request to get user location timed out.")
+      break;
+    case error.UNKNOWN_ERROR:
+      console.log("An unknown error occurred.")
+      break;
   }
 }
 
+
+componentDidMount() {
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition((position) => {
+      this.setState({
+        userLng: position.coords.longitude,
+        userLat: position.coords.latitude});
+
+    }, this.catchGeoLocationError
+);
+  } else {
+    this.openGeoLocationModal();
+  }
+}
+
+
+render() {
+  return (<div className="App">
+            <nav className="App-navigation">
+              <h1 className="App-title">Trash App</h1>
+              <Link to="/">Welcome</Link>
+              <Link to="/browse/">Browse</Link>
+              <Link to="/recycle/">Recycle</Link>
+            </nav>
+
+            <div className="App-mainContent">
+              <Switch>
+                <Route exact="exact" path='/' component={Welcome}/>
+                <Route exact="exact" path='/browse/' component={Browse}/>
+                <Route exact="exact"
+                       path='/recycle/'
+                       render={(routeProps) => (<RecyclePage {...routeProps}
```

Comment:
routeProps was new to me. I learned a coding :)

Reply:
oh yeah this was confusing. we should have gone over this in class imo